### PR TITLE
Add Hbase metrics Receiver

### PR DESCRIPTION
**Pull request by Mrod1598**
Add Hbase metrics Receiver. There were some requested metrics additions upstream, Including adding `mean`, `min`, and `median` to all operations.

### CODE REVIEW

File: apps/hbase.go
Comment (qingling128) [inline apps/hbase.go:48]: Can we refactor this part a bit to use the common util by https://github.com/GoogleCloudPlatform/ops-agent/pull/370?

File: integration_test/third_party_apps_data/applications/hbase/centos_rhel/install
Comment (qingling128) [inline integration_test/third_party_apps_data/applications/hbase/centos_rhel/install:7]: Chances are a bit higher that users will use the default package manager (e.g. yum, apt-get, zyypper) to install a package instead of a pinned version of the application.

Can we adjust it to do something like `yum install hadoop` instead? Similar requests for the other `install` scripts.

File: docs/hbase.md
Comment (qingling128) [inline docs/hbase.md:25]: It's preferred for the receiver and pipeline IDs to stay the same as the receiver type in the sample configuration. E.g.  "s/{APP}_metrics/{APP}", "s/{APP}_pipeline/{APP}"

File: docs/hbase.md
Comment (qingling128) [inline docs/hbase.md:27]: Remove `collection_interval` as it's optional now.

File: confgenerator/testdata/valid/linux/metrics-receiver_hbase/input.yaml
Comment (qingling128) [inline confgenerator/testdata/valid/linux/metrics-receiver_hbase/input.yaml:22]: Remove `collection_interval` as it's no longer required?

File: integration_test/third_party_apps_data/applications/hbase/sles/post
Comment (qingling128) [inline integration_test/third_party_apps_data/applications/hbase/sles/post:4]: For a metrics side only PR, the Fluent Bit submodule should not be updated.

image

File: integration_test/third_party_apps_data/applications/hbase/centos_rhel/install
Comment (Mrod1598) [reply integration_test/third_party_apps_data/applications/hbase/centos_rhel/install:7]: There doesn't seem to be one for hbase either.

File: integration_test/third_party_apps_data/applications/hbase/sles/post
Comment (Mrod1598) [reply integration_test/third_party_apps_data/applications/hbase/sles/post:4]: Changes Removed

File: confgenerator/testdata/valid/linux/metrics-receiver_hbase/input.yaml
Comment (Mrod1598) [reply confgenerator/testdata/valid/linux/metrics-receiver_hbase/input.yaml]: Removed

File: docs/hbase.md
Comment (Mrod1598) [reply docs/hbase.md]: Removed

File: docs/hbase.md
Comment (Mrod1598) [reply docs/hbase.md]: Updated

File: apps/hbase.go
Comment (Mrod1598) [reply apps/hbase.go]: Updated

File: docs/hbase.md
Comment (quentinmit) [inline docs/hbase.md:39]: Document the values of the `state` label?

File: docs/hbase.md
Comment (quentinmit) [inline docs/hbase.md:3]: Please explain in the docs how the cluster metrics are collected (does it only collect the local region server's metrics? or metrics for all the region servers in the cluster?).

File: docs/hbase.md
Comment (quentinmit) [inline docs/hbase.md:43]: These metrics all show a `region_server` label but the scoping doc says that label is a resource label and so won't be presentt.

File: integration_test/third_party_apps_data/applications/hbase/centos_rhel/install
Comment (qingling128) [reply integration_test/third_party_apps_data/applications/hbase/centos_rhel/install:7]: Got it. Guess we have to maintain this piece of code then. NVM

File: docs/hbase.md
Comment (Mrod1598) [reply docs/hbase.md:43]: Due to limitation in the JMX collector, it does not distinguish between normal and resource attributes. So, it will be visible to the customer as a normal attribute. This may change down the round as the collector evolves.

File: docs/hbase.md
Comment (Mrod1598) [reply docs/hbase.md]: Added

File: docs/hbase.md
Comment (Mrod1598) [reply docs/hbase.md:41]: Added

File: docs/hbase.md
Comment (quentinmit) [reply docs/hbase.md:45]: We should strip the label using OT config then, if the label always has a single value.

File: docs/hbase.md
Comment (quentinmit) [inline docs/hbase.md:5]: Please try to make this paragraph clearer to explain that it only applies to master nodes; I read it as saying "you should only configure the receiver on one node", which isn't true - you should configure it on "one master node and  all region server nodes".

If the "if the node being monitored is indeed the Master" is accurate, can't you just configure it on all of the nodes and there just won't be master metrics on the replicants until the replicant becomes the master?

File: docs/hbase.md
Comment (quentinmit) [inline docs/hbase.md:9]: I don't think this "By default" sentence is accurate - your integration test has to manually enable JMX, and also the default port seems to be :10101 and not :8050.

**Comment by qingling128** [comment]
BTW: There is some on-going conversation in the `Regional metrics and master metrics` section of the `Duplicate / Overlapping Metrics Issue from Clustered Applications` doc that is related. Just shared it.

File: docs/hbase.md
Comment (qingling128) [reply docs/hbase.md:5]: Hbase has some pending discussions going on in the `Category 3 - Regional metrics and master metrics` section of the `Duplicate / Overlapping Metrics Issue from Clustered Applications` doc.

The phrasing here might be adjusted based on the decision. We'll need to resolve that conversion before releasing this.

**Review by qingling128 — APPROVED** [review]
LGTM modulo 2 minor comments.

After syncing to master, let's trigger an integration tests for a sanity check before merging, as the last green integration test run had been there for a while.

File: docs/hbase.md
Comment (Mrod1598) [reply docs/hbase.md:24]: Updated

File: docs/hbase.md
Comment (Mrod1598) [reply docs/hbase.md]: Updated

**Pull request merged**
